src/charm.py

```diff
@@ -118,6 +122,19 @@ def _on_generate_action(self, event: ActionEvent):
             "response": response["response"],
         })
 
+    def _on_pull_action(self, event: ActionEvent):
+        if not "model" in event.params:
+            event.fail("Parameter \"model\" was not provided, it is required.")
+            return
+        model = event.params["model"]
+
+        event.log(f"Downloading model {model}…")
+        requests.post(
+            f"http://localhost:{self._charm_state.port}/api/pull",
+            json={"name": model}
+        )
+        event.log("Done.")
```

Comment:
You should wrap the request with `try` and use `event.fail`.

Reply:
Done here: https://github.com/Thinking-Dragon/ollama-juju-charm/pull/10